src_kt/java/arcs/util/random.kt

```diff
@@ -0,0 +1,43 @@
+package arcs.util
+
+import kotlin.math.pow
+import kotlin.random.Random as KotlinRandom
+
+/**
+ * Instance of [kotlin.random.Random] to use arcs-wide.
+ *
+ * TODO: Consider seeding the random object ourselves. Unfortunately kotlin.system.getTimeMillis
+ *   isn't in the common kotlin library. This will mean we need to make this an `expect` and
+ *   implement it for JVM, JS, and WASM targets. Bonus points for using SecureRandom when running in
+ *   JVM.
+ */
+val Random: KotlinRandom
+  get() = RandomBuilder(null)
+
+/**
+ * Generator of a kotlin random class instance.
+ *
+ * This variable is configurable so as to make it possible for tests to make predictable 'random'
+ * behavior possible.
+ */
+var RandomBuilder: (seed: Long?) -> KotlinRandom = fn@{ seed ->
+  val globalRandom = globalRandomInstance
+
+  @Suppress("IfThenToElvis") // Because it's more readable like this.
+  if (globalRandom != null) {
+    // If we've already initialized the global random instance, use it.
+    return@fn globalRandom
+  } else {
+    // Looks like we need to initialize it still, so - depending on whether or not we have a seed,
+    // either return a seeded KotlinRandom, or the default.
+    return@fn (seed?.let { KotlinRandom(seed) } ?: KotlinRandom.Default)
```

Comment:
Can you explain what's going on here for me in this line? Lots of Kotlin magic I don't understand.

Reply:
This if/else reads in english(-ish?) as:
```
if the global random instance has been configured:
  return it (at the named closure: `fn`)
else 
  If seed is non-null: 
    return the result of `KotlinRandom(seed)`(at the closure)
  else:
    return the default implementation 
  and before returning anything, update the random instance to the result of the previous conditional.
```

And the whole closure is a function object we're assigning to a mutable variable: `RandomBuilder`